FatPkg/EnhancedFatDxe/Fat.h

```diff
@@ -84,6 +84,19 @@ SPDX-License-Identifier: BSD-2-Clause-Patent
 #define FAT_FATCACHE_GROUP_MIN_COUNT      1
 #define FAT_FATCACHE_GROUP_MAX_COUNT      16
 
+// For cache block bits, use a UINT64
+#define DIRTY_BLOCKS          UINT64

```

Comment:
1. using typedef instead of #define for types: typedef UINT64 DIRTY_BLOCKS;
2. The concept is one cache tag contains multiple physical blocks.
   So, how about using UINT8 instead of UINT64? Nevertheless, you need an array of either UINT64[2] or UINT8[16].
   #define NUM_BLOCKS_PER_TAG  ((1 << FAT_DATACACHE_PAGE_MAX_ALIGNMENT) / (1 << MIN_BLOCK_ALIGNMENT))
   UINT8 IsBlockDirty[NUM_BLOCKS_PER_TAG/sizeof (UINT8)];

Reply:
I can change to a typedef, I am on vacation for the week, let me come back to point 2

Reply:
@niruiyu, we are already using an array in the CACHE_TAG struct, see below. Using a UINT64 should be fine here, it allows some faster math than UINT8 that the compiler can insert.